meta-lxatac-software/recipes-devtools/python/python3-labgrid/configuration.yaml

```
@@ -103,3 +103,17 @@ lxatac-usb-ports-p{{idx}}.{{if}}:
       '@ID_PATH': '{{sysfs}}:1.{{if}}'
 {% endfor %}
 {% endfor %}
+
+
+## Interal TFTP Server
+tftp:
+  TFTPProvider:
+    internal: "/srv/tftp/"
+    external: "/"
+
+## Internal HTTP Server
+http:
+  HTTPProvider:
+    internal: "/srv/www/"
+    external: "/srv/"
```

Should the `external` entry refer to an URL instead? That is how I read the [documentation](https://labgrid.readthedocs.io/en/latest/configuration.html#httpprovider).

---

I was not aware, that one *can* use it with an URL.

In my configuration I am using it as shown here.
Then, my strategy uses it like this:
```
url = self.http.stage(self.target.env.config.get_image_path("ram_boot"))
# url: http://192.168.1.10/images/openwrt-24.10.2-ramips-mt7621-genexis_pulse-ex400-initramfs-kernel.bin
p = urllib.parse.urlparse(url)
# p: ParseResult(scheme='http', netloc='192.168.1.10', path='/images/openwrt-24.10.2-ramips-mt7621-genexis_pulse-ex400-initramfs-kernel.bin', params='', query='', fragment='')
```

If we want to add an host to the export we could use the LXA TACs hostname and then rely on DNS.

---

Exactly. I think that's how it's meant to be used. The documentation could be better.

```suggestion
    external: "http://{{hostname}}/srv/"
```

See: https://labgrid.readthedocs.io/en/latest/configuration.html#templating-the-exporter-configuration

---

Ok, in this case the `HTTPProvider` example should be updated to use the `{{hostname}}`-template.